assets/konnectivity/konnectivity-agent-control-plane.yaml

```diff
@@ -0,0 +1,94 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: konnectivity-agent
+  labels:
+    app: konnectivity-agent
+    clusterID: {{ .ClusterID }}
+spec:
+  progressDeadlineSeconds: 600
+  replicas: 3
+  revisionHistoryLimit: 10
+  selector:
+    matchLabels:
+      app: konnectivity-agent
+  strategy:
+    rollingUpdate:
+      maxSurge: 0
+      maxUnavailable: 1
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app: konnectivity-agent
+        clusterID: {{ .ClusterID }}
+    spec:
+      automountServiceAccountToken: false
+{{- if .MasterPriorityClass }}
+      priorityClassName: {{ .MasterPriorityClass }}
+{{- end }}
+      containers:
+      - name: konnectivity-agent
+{{- if .KonnectivitySecurityContext }}
+{{- $securityContext := .KonnectivitySecurityContext }}
+        securityContext:
+          runAsUser: {{ $securityContext.RunAsUser }}
+{{- end }}
+        image: {{ imageFor "apiserver-network-proxy" }}
+        imagePullPolicy: IfNotPresent
+        command: ["/bin/proxy-agent"]
+        args: [
+          "--logtostderr=true",
+          "--ca-cert=/etc/konnectivity/ca/ca.crt",
+          "--agent-cert=/etc/konnectivity/agent/tls.crt",
+          "--agent-key=/etc/konnectivity/agent/tls.key",
+          "--proxy-server-host=konnectivity-server.{{ .Namespace }}.svc",
+          "--proxy-server-port={{ .KonnectivityAgentPort }}",
+          "--health-server-port={{ .KonnectivityAgentHealthPort }}",
+          "--agent-identifiers=ipv4={{ .OpenShiftAPIClusterIP }}&ipv4={{ .OauthAPIClusterIP }}",
+          "--keepalive-time=30s",
```

Comment:
Per https://github.ibm.com/alchemy-containers/armada-ansible/pull/12017/, should we increase the timeout?  If the IKS value is preferred then please work with the HyperShift folks to do the same.

Reply:
As I remember, in IKS, we wanted 30secs, but you told us that some containerd keepalive time is 40 sec, so let's have it 40 secs. Both times are fine, we don't see problem - only the default value is too high.
I would leave as is for now. Thanks!